### DISCUSSION TITLE
Scrolling to input container on error

### DISCUSSION THREAD

**Discussion by specialdoom** [react-hook-form/react-hook-form#7256] [Q&A]
Can somebody explain this behaviour?

I have an `input` wrapped inside a `div` and the field `ref` from `Controller` passed to `div` instead of `input`?

Can somebody explain why, on error, the div element it's not focused? If I move the `ref` to input than everything it's perfect.

I'm asking for a case where I really need the `ref` on the container and not the actual `input`.

Steps:
A. `ref` on `div`
  1. Scroll to end on page
  2. Click on submit
  3. No `div` focus on error

B. `ref` on `input`
  1. Scroll to end on page
  2. Click on submit
  3. `input` focus on error

CodeSandBox: https://codesandbox.io/s/smoosh-monad-w1onc?file=/src/App.js

**Comment by JamesWong1999** [comment]
react hook form using `.focus()`, div is not going to work.

**Reply by davidalvarezr** [reply]
In your A case, if you want your div to  be focused, just add the attribute tabIndex={0} to the div @specialdoom
